Разметка всех страниц

Krit74 opened the pull request:
---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/34/user/1363607/tasks/4)

Krit74 committed:
добавляет обертку для всей страницы

Krit74 committed:
добавляет логотип

Krit74 committed:
добавляет навигацию по сайту

Krit74 committed:
добавляет пользовательское меню и ссылку на модальное окно - header готов

Krit74 committed:
исправляет ошибку в написании класса menu

Krit74 committed:
добавляет первую секцию с приветствием

Krit74 committed:
добавляет первый блок преимуществ

Krit74 committed:
добавляет второй блок с картинкой

Krit74 committed:
добавляет 3,4 и 5 блок копированием 5 с удалением subtytlle

Krit74 committed:
копирует 1 и2 блоки с правками классов и текста по макету

Krit74 committed:
исправляет модификаторы в последних 3-х блоках

Krit74 committed:
добавляет секцию поиска с переходом  на модальное окно

Krit74 committed:
добавляет секцию рассылок

Krit74 committed:
исправляет класс у кнопки хедера и добавляет модификатор

Krit74 committed:
добавляет подвал

Krit74 committed:
Создает страницу catalog.html

Krit74 committed:
вносит правки и добавляет классы в лого и меню

Krit74 committed:
добавляет заголовок и хлебные крошки

Krit74 committed:
добавляет секцию с фильтрами

Krit74 committed:
исправляет название класса страницы

Krit74 committed:
добавляет контрольную панель

Krit74 committed:
добавляет карточку каталога

Krit74 committed:
добавляет копированием остальные карточки

Krit74 committed:
испправляет модификатор кнопки во второй карточке

Krit74 committed:
добавляет блок с пагинацией

Krit74 committed:
добавляет сортировку

Krit74 committed:
добавляет модификатор к кнопке "подписаться"

Krit74 committed:
добавляет модификатор к кнопке "подписаться" в index.html

Krit74 committed:
исправляет название файла

Krit74 committed:
исправляет модификатор кнопки в шапке

mates4352 commented on index.html:
```diff
@@ -1,12 +1,20 @@
+</head>
+
+<body>
+  <div class="container-page">
```
Название класса предпочтительней было бы назвать main-page

mates4352 commented on index.html:
```diff
@@ -1,12 +1,20 @@
+
+<body>
+  <div class="container-page">
+    <div class="wraper-page">
```
Название класса предпочтительней было бы назвать main-page-container

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+        <nav class="menu">
+          <ul class="menu-list">
+            <li class="menu-item">
+              <a class="menu-link" href="">Главная</a>
```
Для атрибута href в ссылке если нет ссылки ставится заглушка href="#"

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+              <a class="menu-link" href="">Гостиницы</a>
+            </li>
+          </ul>
+          <ul class="menu-list menu-user">
```
Спорный момент насчет списка так как элементы имеют разный контент, если делать то кнопку тоже нужно добавлять в тег li. В тег ul можно добавлять теги li и script другие теги не проходят валидация.

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+            <li class="greetings-description-item">
+              <h2 class="greetings-deescription-title">Седона — небольшой городок в Аризоне,
+                заслуживающий большего!</h2>
+              <b class="greetings-deescription-text">Рассмотрим 5 причин, по которым Седона круче, чем Гранд-Каньон!</b>
```
Правильность использование тега b заключается добавление в него ключевых слов который важен для данной страницы в этом случае это тег p

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+      <main class="main-index">
+        <h1 class="visually-hidden">Седона</h1>
+        <section class="greetings">
+          <img class="greetings-imadges" src="#" alt="Добро пожаловать в великолепную Седону" />
```
В начале каждого alt я бы добавил слово 'Изображение' а после уже описание этого изображение

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+        <h1 class="visually-hidden">Седона</h1>
+        <section class="greetings">
+          <img class="greetings-imadges" src="#" alt="Добро пожаловать в великолепную Седону" />
+          <ul class="greetings-description-list">
```
Здесь нет необходимости в теге ul так как контент не является списком элементов

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+          <h2 class="visually-hidden">Преимущества</h2>
+          <ul class="advantages-grid">
+            <li class="advantages-grid-item advantages-grid-item__blue-bassic3">
+              <h3 class="advantages-grid-title">Настоящий городок</h3>
```
В li идет article в нем уже весь контент.
article это самодостаточный контент, например если мы сейчас уберем его суть сайта не измениться

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+          <ul class="advantages-grid">
+            <li class="advantages-grid-item advantages-grid-item__blue-bassic3">
+              <h3 class="advantages-grid-title">Настоящий городок</h3>
+              <b class="advantages-grid-subtitle">— №1 —</b>
```
Это просто span

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+        <section class="search">
+          <h2 class="visually-hidden">Поиск гостиниц</h2>
+          <h3 class="search-title">Заинтересовались?</h3>
+          <b class="search-text">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в
```
Здесь тег p если нужно подчеркнуть важности определённого участка текста то можно внутри p вставить тег strong

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+          <h3 class="search-title">Заинтересовались?</h3>
+          <b class="search-text">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в
+            Седоне</b>
+          <button class="button button-search" type="submit">Поиск гостиниц в Седоне</button>
```
Это ссылка на страницу каталог

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+        </section>
+        <section class="newsletter">
+          <h2 class="visually-hidden">Подписка на рассылки</h2>
+          <h3 class="newsletter-title">Получайте лучшие предложения</h3>
```
В разметке нет данного контента. Eсли стоят 2 заголовка рядом их следует обернуть в тег hgroup чтоб они представляли одну сематическую сущность

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+        <section class="newsletter">
+          <h2 class="visually-hidden">Подписка на рассылки</h2>
+          <h3 class="newsletter-title">Получайте лучшие предложения</h3>
+          <b class="newsletter-text">Только полезная информация и никакого спама, честное бойскаутское!</b>
```
Тег p

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+          <b class="newsletter-text">Только полезная информация и никакого спама, честное бойскаутское!</b>
+          <form class="newsletter-form" action="https://echo.htmlacademy.ru/" method="post">
+            <input class="field" placeholder="Ваш e-mail" type="email" name="newsletter-email" required>
+            <button class="button newsletter-button butten__blue" type="submit">Подписаться</button>
```
Тег button по умолчанию имеет type="submit" можно ставить можно нет

Krit74 replied:
пока оставлю... 
на Лайве Андрей наоборот сказал, что лучше ставить в tupe submit
один из вопросов которые к консультации

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+        </section>
+      </main>
+      <footer class="footer-page">
+        <section class="footer-social">
```
Не имеет смысла вкладывать в footer тег section.
footer- тег который описывает контент как не имеющий связь с основным контентом страницы поэтому обычно все любят говорить что это подвал документа потому-что в низу сайта лежит информация которая грубо говоря не нужна.
section - тег описывает основной контент страницы или его дополняет.
Поэтому нету смысла вкладывать section в footer

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+      <footer class="footer-page">
+        <section class="footer-social">
+          <h2 class="visually-hidden">Контакты</h2>
+          <ul class="footer-social-list">
```
Желательно добавить тег nav для иконок

mates4352 commented on catalog.html:
```diff
@@ -0,0 +1,265 @@
+      <main class="main-inner">
+        <section class="selection">
+          <h1 class="selection-title">Гостиныцы Седоны</h1>
+          <ul class="breadcrumbs">
```
Используем тег ol в данном случае на важен порядок элементов ul это не упорядоченный список элементов

mates4352 commented on catalog.html:
```diff
@@ -0,0 +1,265 @@
+              </ul>
+            </fieldset>
+          </form>
+          <p>здесь должен быть ползунок, но я не знаю как его сделать</p>
```
Пока не делай, это будет в следующих уроках

mates4352 commented on catalog.html:
```diff
@@ -0,0 +1,265 @@
+          </form>
+          <p>здесь должен быть ползунок, но я не знаю как его сделать</p>
+          <button class="button butten-filter butten-filter__blue " type="submit">Примменить</button>
+          <button class="button butten-filter butten-filter__hidden" type="submit">Сбросить</button>
```
type="reset"

mates4352 commented on catalog.html:
```diff
@@ -0,0 +1,265 @@
+          <button class="button butten-filter butten-filter__blue " type="submit">Примменить</button>
+          <button class="button butten-filter butten-filter__hidden" type="submit">Сбросить</button>
+        </section>
+        <section class="catalog">
```
h2 каталог гостиниц

Krit74 replied:
странно, я добавил скрытый заголовок, но комент остался не отмеченным

mates4352 commented on catalog.html:
```diff
@@ -0,0 +1,265 @@
+            <select class="select-control" aria-label="сортировка">
+              <option value="cheap">Сначало дешевые</option>
+            </select>
+            <a class="button buttn-small" href="#">
```
это список кнопок

mates4352 commented on index.html:
```diff
@@ -1,12 +1,143 @@
+              </a>
+            </li>
+          </ul>
+          <section>
```
Не закрыл тег

mates4352 commented on catalog.html:
```diff
@@ -0,0 +1,265 @@
+          </div>
+          <ul class="catalog-list">
+            <li class="catalog-card">
+              <img class="catalog-card-imge" src="#" alt="" />
```
тег article потом можно добавить 
figure img figcaption
Предлагаю вынести данный блок с карточками на консультация

Krit74 replied:
хорошо

Krit74 committed:
исправляет название класса страницы

Krit74 committed:
добавляет в href заглушку #

Krit74 committed:
перемещает кнопку в список menu-user

Krit74 committed:
исправляет тег b на p

Krit74 committed:
исправляет описание alt

а разве скринридеры не будут читать такое описание как "картинка изображение ......." ?

Krit74 committed:
удаляет лишние теги

Krit74 committed:
добавляет тег article в карточке преимуществ

Krit74 committed:
исправляет тег b на span

Krit74 committed:
исправляет тег b на p

Krit74 committed:
заменяет кнопку на ссылку

Krit74 committed:
удаляет скрытый заголовок и заменяет теги h3 на h2 b на p

Krit74 committed:
удаляет тег section, добавляет тег nav для social

тег nav - навигация, я его больше с меню отождествляю чем со списком соцсетей

Krit74 committed:
исправляет type в кнопке "сбросить"

Krit74 committed:
добавляет скрытый заголовок в секцию "Каталог"

Krit74 committed:
добавляет к кнопкам вида страниц тег ul

Krit74 committed:
делает исправления в секции newsletter и footer аналогичные index.html
